### ISSUE TITLE
not converge, related problem

### ISSUE THREAD

**Issue by cristina510** [kyu-sz/WPAL-network#17]
Hello.  I run this WPAL-NETWORK code from github.There still exists a problem which confuses me and I'm wondering if you could give me some advice.

      When I run the training step, I first encountered the error "check failed：bottom[0]->count ==bottom[1]->count“  in SigmoidCrossEntropyLossLayer.  I solved it in this way:
      I vertify this line in config.py:
      _C.NUM_ATTR = 53 
     I change the number of attributes 53 to 92 and there seems no such error exists.

     However, I run the training step and the loss is about 68. Then the loss maintains NAN.
     I tried to change the base_lr:0.01 smaller or batch_size bigger, but the loss turns to change like this: 68->66->70->55->30->63,,and never seems to converge.

    Could you please give me some advice? Thanks very much.

**Comment by kyu-sz** [comment]
Hi, thank you for your interest in my project.
As for the attribute number not matching problem, make sure the script, the network, the dataset and the configuration python script all share a same setting of attribute number.
As for the convergence problem, yes this network is slow in convergence, but it should finally converge with loss at about 5. Please check your training data and other settings.
And there is a new version of this project with lots of improvement, as referred in readme. It is strongly recommended that you start your work based on that version.
Good luck!

**Comment by cristina510** [comment]
Thanks for your reply. I've trained the network for about 913,000 iterations and **the loss fluctuates between 9.8 to 15** for 2 days' training, while you said 'it should finally converge with loss at about 5'. It appears that there's something wrong.
And I also test it using the snapshoot 910000_iter.caffemodel. The results seems not well. The Acc and Prec are both about 0.2 , Rec is 0.91 and mA=7.8...
I'm wondering if  you could give me some advice on this. Thanks a lot for your time.

**Comment by kyu-sz** [comment]
Increasing batch size as large as possible might help improving convergence as well as performance.
However, a model with loss at about 10 should display some ability of attribute recognition, and a mA of 7.8 is apparently unusual. I guess you should take a look at your training and test dataset, and make sure they are read in and utilized in the expected manner.

**Comment by cristina510** [comment]
Thanks for your prompt reply. I truly appreciate your time:)
I've been reading the code and **check if the dataset are properly used.**  However, I'm still stuck with the problem. I checked the following items 
(1) The RAP dataset is unzipped to the directory ./data/dataset/, as is written on github.
(2) I used  the file train_rap_0.sh in the floder INCEPTION_BN_SPP, and then used  train_rap_0.sh. So is it the right way to use the files?

**Comment by cristina510** [comment]
It's true that there's something wrong with the test codes.It seems I've got the correct results. Thanks~~

**Comment by Awadhesh42** [comment]
@cristina510 can you please share your trained model as I am unable to train it due to limitation of my GPU. 
I am using GOOGLENET and after 4 days of training I am still getting loss of 20%.

**Comment by Awadhesh42** [comment]
@cristina510 also i am getting some error while testing can you please help me to debug it.
I am using this project as a part of my university course work. I have to finish it as quickly as I can.
Please help me!

**Comment by XipengY** [comment]
@cristina510  Excuse me, can you share the problems you found in test code? I've got the same error, Acc=0.18 Prec=0.19 Rec=0.886, mA=0.77

**Comment by kclch** [comment]
@cristina510 Hello, I am also studying this project recently. Have you trained the model?
